monai/deploy/operators/nii_data_loader_operator.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021-2022 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import SimpleITK
+
+import monai.deploy.core as md
+from monai.deploy.core import DataPath, ExecutionContext, InputContext, IOType, Operator, OutputContext
+
+
+@md.input("image_path", DataPath, IOType.DISK)
+@md.output("image", np.ndarray, IOType.IN_MEMORY)
+class NiftiDataLoader(Operator):
+    """
+    This operator reads a nifti image, extracts the numpy array and forwards it to the next operator
+    """
+
+    def compute(self, op_input: InputContext, op_output: OutputContext, context: ExecutionContext):
+        input_path = op_input.get().path
+        image_np = self.convert_and_save(input_path)
```

Comment:
The op_input path is typically a folder, even though file path is also possible depending on how the input is set up. It is important that the path itself is checked for what type it is.

Also, in the case of folder, there has to logic to scan the file(s) for supported types, e.g. nii, nii.gz, mhd, etc. The monai core has the `LoadImage` class that has a set of supported readers, including nii, and wonder why not simple use it in the application, or make use of it here.

Reply:
Yes monai core has the logic for `LoadImage`. But, it is also very specific in terms of how it wants to handle images. Now it is using metadata, there is this idea of channel first etc. I wanted a very raw example, which someone can look and manipulate more easily if needed. We will talk about it today. 
Thanks @MMelQin

Reply:
`LoadImage` shouldn't manipulate the image data at all and leaves that to subsequent transforms. It should preserve the Nifti header in the metadata so you can use it later especially for saving output images back to the same format as the input. It doesn't however take as input single file names so to scan a directory we typically use `glob` to get a list of files.

Reply:
@MMelQin I understand that the op_input path is typically a folder. But that is a valid point for dicom images. The nifti images are just files. In a typical scenario, all the nifti images will be contained in a single folder so it makes sense to use op_input path for a file instead of a folder. 

@ericspod  and @MMelQin 
Yes `LoadImage` can be used for loading the images. However, in the monai environment the channels are considered first by default. and there might be use cases where that is not the case. In addition, I thought it will be nice to have an example where you can show that monai deploy operators do not necessarily depend on the monai core components and it is possible to play around with other readers and demonstrate the flexibility it offers. 
But if you think, it is not particularly useful please go ahead with the changes.

Reply:
Channels are added first because that's what Pytorch wants when considering what the dimensions of a tensor represent. What we often do is use `Orientation/Orientationd` transforms to change the orientation of the image based on what the Nifti header says so that the spatial dimensions always represent the same thing, like [here](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb). If that's the kind of situation in which you wouldn't want to add a channel dimension then I'd say the use of these transforms together is the correct solution. 

I'm obviously biased but I'd always recommend using MONAI code rather than other libraries for doing the same thing.